analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -150,6 +189,117 @@ class SpdxDocumentFile(
             }
         }
 
+    /**
+     * Maps a given identifier on an spdx package or external document reference
+     * @param doc spdx document containing all packages or external document references the id could refer to
+     * @param source the identifier to be found in the spdx document
+     */
+    private fun getSpdxPackageFromId(
+        doc: SpdxDocument,
+        source: String
+    ): SpdxPackage {
+        val fromPackage = doc.packages.singleOrNull { it.spdxId == source }
+        if (fromPackage != null) {
+            return fromPackage
+        }
+        val externalDocumentReference = doc.externalDocumentRefs.singleOrNull {
+            it.externalDocumentId == source.split(":").first()
+        }
+            ?: throw IllegalArgumentException("No single package or externalDocumentRef with " +
+                    "source ID '$source' found.")
+
+        val externalDocumentReferenceToSpdxPackage = externalDocumentReferenceToSpdxPackage(externalDocumentReference)
+        packagesFromExternalDocumentReferences.add(externalDocumentReferenceToSpdxPackage)
+        return externalDocumentReferenceToSpdxPackage
+    }
+
+    /**
+     * Maps [SpdxExternalDocumentReference.spdxDocument] to a package
+     * @param externalDocumentReference external document reference containing a spdx document with a uri to a spdx
+     * document that can be mapped to a [SpdxPackage]
+     * @throws IllegalArgumentException
+     * @returns spdx package the spdx document uri of the externalDocumentReference was referring to
+     */
+    private fun externalDocumentReferenceToSpdxPackage(externalDocumentReference: SpdxExternalDocumentReference):
+            SpdxPackage {
+        val externalSpdxDocument = getFileFromExternalDocumentReference(externalDocumentReference)
+
+        if (listOf("json", "yaml", "yml").contains(externalSpdxDocument.extension)) {
```

Comment:
@sschuberth I am surprised you would suggest catching and re-throwing the same exception, as I thought that wasn't clean code (if not for a special case within several possible exceptions to be caught). But since it would just be one exception, I would not catch and re-throw here.

Reply:
IMO re-throwing the same exception is fine if you want to amend the exception message with more context.

Anyway, we should leave the extension check to the implementation of `SpdxModelMapper.read()`.

Reply:
I had to stop review at this line, will continue tomorrow.